engine/storage/datamotion/src/org/apache/cloudstack/storage/motion/StorageSystemDataMotionStrategy.java

```diff
@@ -172,6 +195,12 @@ private void validate(SnapshotInfo snapshotInfo) {
         }
     }
 
+    private boolean usingBackendSnapshotFor(SnapshotInfo snapshotInfo) {
+        String property = getProperty(snapshotInfo.getId(), "takeSnapshot");
+
+        return Boolean.parseBoolean(property);
+    }
+
     private Void handleCreateTemplateFromSnapshot(SnapshotInfo snapshotInfo, TemplateInfo templateInfo, AsyncCompletionCallback<CopyCommandResult> callback) {
```

Comment:
Hi @mike-tutkowski ,
Consider changing "Void" to "void" and removing the "return null;" at the end of this method.
Another suggestion is to create a single method, if possible, for the lines 260 to 266 and 363 to 371, this would avoid some duplication of code.